### ISSUE TITLE
Power and Antenna parts not recognized for contract generation

### ISSUE THREAD

**Issue by symmeclept** [RemoteTechnologiesGroup/RemoteTech#701]
OS: Windows 7 x64
KSP: 1.2.1 x64

Problem: With RemoteTech 1.8.1 installed, contracts requiring parts with module Power or Antenna do not generate. Might be related to contracts not completing, as well.

[output_log](https://gist.github.com/symmeclept/bc24618752d770e3df7c26987eac363d)

[mod list](https://gist.github.com/symmeclept/39a9320324139af88acd2bbe002dc485)

[persistent.sfs](https://gist.github.com/symmeclept/e5cf8dde0402fcb95590a7cd7b6fd82f)

**Comment by KSP-TaxiService** [comment]
I tested this fix but found it was not working. The `GetContractObjectiveType()` and `CheckContractObjectiveValidity()` in `ModuleRTDataTransmitter.cs` were never called at all times. Our RT codes use `ModuleRTDataTransmitter.cs` class in an weird way:
```
if (node.HasNode("TRANSMITTER"))
 {
      RTLog.Notify("ModuleRTAntenna: Found TRANSMITTER block.");
      mTransmitterConfig = node.GetNode("TRANSMITTER");
      mTransmitterConfig.AddValue("name", "ModuleRTDataTransmitter");
      ....
}
```

On my own side, I fixed it by moving the interface and its method implementations to `ModuleRTAntenna.cs` and tested in KSP. Both methods were called and I can get satellite contracts. But I did not commit this change because I need some opinion on this change.
